src/routers/resource_router.py

```diff
@@ -485,13 +486,35 @@ def register_resource(
 
         return register_resource
 
-    def create_resource(self, session: Session, resource_create_instance: SQLModel):
+    def create_resource(
+        self,
+        session: Session,
+        resource_create_instance: SQLModel,
+        user: KeycloakUser = Depends(get_user_or_raise),
+    ):
         """Store a resource in the database"""
         resource = self.resource_class.from_orm(resource_create_instance)
         deserialize_resource_relationships(
             session, self.resource_class, resource, resource_create_instance
         )
         session.add(resource)
+        session.flush()
+
+        if user.is_connector:
+            # 1. Trust connector's input, do not override platform/platform_resource_identifier
+            # error message is raised if the connector does not provide these fields.
+            pass
+        # 2. Normal user: must NOT provide platform/platform_resource_identifier
+        else:
+            if resource.platform is not None or resource.platform_resource_identifier is not None:
+                raise HTTPException(
+                    status_code=status.HTTP_400_BAD_REQUEST,
+                    detail="No permission to set platform or platform_resource_identifier fields.",
+                )
+            # Set these fields as required for normal users
+            resource.platform = PlatformName.aiod
+            resource.platform_resource_identifier = resource.aiod_entry_identifier
```

Comment:
```suggestion
            resource.platform_resource_identifier = resource.identifier
```
Let's keep it the resource identifier. With the migration already in place, it's also unique across the platform. And in the future we might match `aiod_entry` identifier to match it, but for now `aiod_entry` is only used internally so we can avoid exposing it here.